### DISCUSSION TITLE
Clerk Integrations Guide uses deprecated `supabase.auth.setAuth`

### DISCUSSION THREAD

**Discussion by JohnGemstone** [supabase/supabase#16852] [Contribute to Supabase]
# Improve documentation

## Link 

[Clerk Integrations Guide](https://supabase.com/docs/guides/integrations/clerk#step-5-set-auth-token-with-supabase)

## Describe the problem

Hi! Seems like the clerk integrations guide is outdated. It still uses the setAuth method which is now deprecated.

## Describe the improvement

Clerk tokens could instead be set in the authorization header via a client function like so:
```tsx
const supabaseClient = async (supabaseAccessToken) => {
  const supabase = createClient(
    process.env.NEXT_PUBLIC_SUPABASE_URL,
    process.env.NEXT_PUBLIC_SUPABASE_KEY,
    {
      global: { headers: { Authorization: `Bearer ${supabaseAccessToken}` } },
    }
  );
  // set Supabase JWT on the client object,
  // so it is sent up with all Supabase requests
  return supabase;
};

  ```
Referenced from [Clerk's Docs](https://clerk.com/docs/nextjs/supabase#configure-your-client) + garyaustin via Discord.

**Comment by novemberkilo** [comment]
Can this issue please get some attention - at the very least please take the clerk integration guide down until it is updated. It is incorrect and needlessly causes time and effort to be wasted.

**Comment by gregnr** [comment]
Thanks for reporting this @JohnGemstone and the ping @novemberkilo. Would either of you be interested in opening a small PR for this?

https://github.com/supabase/supabase/blob/5e665120a46524e6b95a18a4c605641731bde8c0/apps/docs/pages/guides/integrations/clerk.mdx?plain=1#L122-L155

**Comment by novemberkilo** [comment]
@gregnr I'm just a few days into working on integrating supabase and clerk into an app so don't feel qualified to update the supabase documentation. 

I think that the folks at clerk have more recent documentation here https://clerk.com/docs/nextjs/supabase and may very well be better placed to contribute a change. @perkinsjr may be able to help?

**Comment by Gingaaa** [comment]
Hey I have worked on around 7-8 company based project, but never countribute to open source, this is my first time can you assign me this task.

**Comment by gregnr** [comment]
@Gingaaa we don't have an assignment process, feel free to start a PR and link it back to this issue!

**Comment by adamchenwei** [comment]
So instead of two places. maybe the official supabase instruction should be removed and just put a link to https://clerk.com/docs/integrations/databases/supabase? Just so a single source of truth is in place.

**Reply by daniello123567** [reply]
hey man , did that doc work for you??
